server/py/framework/db/sqldb/db.py

```diff
@@ -7149,3 +7152,103 @@ def _paginate_query(
             query = query.limit(limit)
 
         return query
+
+    @staticmethod
+    def _find_artifact_latest_tag(session, object_record):
+        """Find the 'latest' tag for an artifact object."""
+        return (
+            session.query(ArtifactV2.Tag)
+            .filter(
+                ArtifactV2.Tag.obj_id == object_record.id,
+                ArtifactV2.Tag.name == mlrun.common.constants.RESERVED_TAG_NAME_LATEST,
+                ArtifactV2.Tag.project == object_record.project,
+            )
+            .one_or_none()
+        )
+
+    @staticmethod
+    def _find_other_latest_in_other_iterations(session, object_record):
+        """Check if there are other 'latest' tags in other iterations for the same producer_id."""
+        return (
+            session.query(ArtifactV2.Tag)
```

Comment:
Here too, since you don't care about the full object but just the fact that it exists, you can use `with_entities` to avoid serializing the full JSON. You can just ask for the `id`.